src/Microsoft.ApiDesignGuidelines.Analyzers/Core/PropertiesShouldNotReturnArrays.cs

```diff
@@ -26,14 +26,26 @@ public abstract class PropertiesShouldNotReturnArraysAnalyzer : DiagnosticAnalyz
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/0fss9skc.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            analysisContext.RegisterSymbolAction(AnalyzeSymbol, SymbolKind.Property);
+
+        }
+
+        private void AnalyzeSymbol(SymbolAnalysisContext context)
+        {
+            var symbol = (IPropertySymbol)context.Symbol;
+            if ((symbol.Type.TypeKind == TypeKind.Array) && !(symbol.IsOverride))
```

Comment:
I think this rule should check a bit more than return typing being array and not overridden such as checking accessibility of the property being "public" or "protected" and its containing type being "public" (probably real public type rather than public type inside of private type - I think we already have library which can check that)

also, we probably can check whether this belong to attribute type or not, and if it is, we can just skip it.
...

Cause
A public or protected property in a public type returns an array.

When to Suppress Warnings
Attributes can contain properties that return arrays, but cannot contain properties that return collections. You can suppress a warning that is raised for a property of an attribute that is derived from the System.Attribute class. Otherwise, do not suppress a warning from this rule.